src/onnx/parse_instancenorm.cpp

```diff
@@ -107,23 +108,32 @@ struct parse_instancenorm : op_parser<parse_instancenorm>
                 });
             n              = 1.0 / std::sqrt(n);
             auto n_literal = info.add_literal(literal{dtype, {n}});
-            mean_bcast     = info.add_common_op("mul", {mean_bcast, n_literal});
             x              = info.add_common_op("mul", {x, n_literal});
         }
-        auto l0              = info.add_instruction(make_op("sqdiff"), x, mean_bcast);
+        auto l0              = info.add_common_op("sqdiff", x, mean);
         auto variance        = info.add_instruction(make_op(reduce_op_name, {{"axes", axes}}), l0);
         auto epsilon_literal = info.add_literal(literal{shape{literal_dtype}, {epsilon}});
-        auto epsilon_bcast =
-            info.add_instruction(make_op("multibroadcast", {{"out_lens", dims}}), epsilon_literal);
-        auto variance_bcast =
-            info.add_instruction(make_op("multibroadcast", {{"out_lens", dims}}), variance);
-        auto l2 = info.add_instruction(make_op("add"), variance_bcast, epsilon_bcast);
+        auto l2              = info.add_common_op("add", variance, epsilon_literal);
+
         auto l3 = info.add_instruction(make_op("rsqrt"), l2);
-        auto l4 = info.add_instruction(make_op("mul"), l1, l3);
-        auto scale_bcast =
-            info.add_instruction(make_op("broadcast", {{"axis", 1}, {"out_lens", dims}}), scale);
-        auto bias_bcast =
-            info.add_instruction(make_op("broadcast", {{"axis", 1}, {"out_lens", dims}}), bias);
+        auto l4 = info.add_common_op("mul", l1, l3);
+
+        // add_common_op not implemented for broadcast op, so use different overloads of make_op.
```

Comment:
why does it need to use broadcast op specifically ? I think it can work with `add_common_op`.

Reply:
No, there's a specific check in insert_common_args() that disallows usage with a single argument if it's a dynamic shape.  I think it's just because dynamic shapes haven't been implemented in that code branch yet--Charlie could probably confirm that.

I just noticed that the exception message for this in common.cpp is extremely misleading--I'll update it.